src/main/java/org/wso2/carbon/connector/ldap/LDAPUtils.java

```diff
@@ -72,6 +80,16 @@ protected static DirContext getDirectoryContext(MessageContext messageContext)
             env.put(LDAPConstants.JAVA_NAMING_LDAP_FACTORY_SOCKET,
                     LDAPConstants.ORG_WSO2_CARBON_CONNECTOR_SECURITY_MYSSLSOCKETFACTORY);
         }
+        env.put(LDAPConstants.COM_SUN_JNDI_LDAP_CONNECT_POOL, String.valueOf(connectionPoolingEnabled));
+        if (StringUtils.isNotEmpty(connectionPoolingProtocol)) {
```

Comment:
Better to add these pooling related parameters only if pooling is enabled.

Reply:
fixed with a9be1264f230227ea720e4b417de783bbf713cc5